CONTRIBUTING.md

```diff
@@ -109,48 +109,356 @@ Check out [building instructions](https://github.com/prestodb/presto/tree/master
+    * The total number of lines modified in a single PR shall not exceed 5000. An exception to this rule is for changes that include checked in code generated files (such as [presto_protocol.cpp](https://github.com/prestodb/presto/blob/master/presto-native-execution/presto_cpp/presto_protocol/presto_protocol.cpp)).
+    * The commits sequence shall be in the dependencies order, not in date created/modified order
+    * Every commit in the PR shall pass all the tests
+      Example: https://github.com/prestodb/presto/pull/12991/commits
+
+* #### Follow the PR template provided
+* #### Backport PR
+    * Must refer to the original PR and provide link in the PR message.
+    * Must refer to original authors in the PR message.
+    * Do NOT change titles or messages of the original commits.
+    * If the original PR contains multiple commits, follow the same structure.
+
+
 ## Pull Request Checklist
 
-- Branch from the master branch and, if needed, rebase to the current master
-  branch before submitting your pull request. If it doesn't merge cleanly with
-  master you may be asked to rebase your changes.
+If you are not an official committer, you must fork from Master and then submit a Pull Request to contribute as an individual.
 
-- If your pull request does not have a reviewer
-  assigned to it after a few days, [ask for a review in the #dev channel in Slack](https://join.slack.com/t/prestodb/shared_invite/enQtNTQ3NjU2MTYyNDA2LTYyOTg3MzUyMWE1YTI3Njc5YjgxZjNiYTgxODAzYjI5YWMwYWE0MTZjYWFhNGMwNjczYjI3N2JhM2ExMGJlMWM).
+To make a contribution:
+
+We use the [Fork and Pull model](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/getting-started/about-collaborative-development-models#fork-and-pull-model)
+- Fork from the master branch
+- If needed, rebase to the current master
+  branch before submitting your pull request
+- If it doesn't merge cleanly with
+  master you may be asked to rebase your changes.
+- If your pull request does not have a reviewer assigned to it after 4 days, [ask for a review in the #dev channel in Slack](https://join.slack.com/t/prestodb/shared_invite/enQtNTQ3NjU2MTYyNDA2LTYyOTg3MzUyMWE1YTI3Njc5YjgxZjNiYTgxODAzYjI5YWMwYWE0MTZjYWFhNGMwNjczYjI3N2JhM2ExMGJlMWM).
 
 - Tests are expected for all bug fixes and new features.
 
-- Make sure your code follows the [code style guidelines](https://github.com/prestodb/presto/blob/master/CONTRIBUTING.md#code-style), [development guidelines](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development) and [formatting guidelines](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting)
+- Make sure your code follows the [code style guidelines](https://github.com/prestodb/presto#code-style), [development guidelines](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development) & [formatting guidelines](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting)
```

Comment:
```suggestion
- Make sure your code follows the [code style guidelines](https://github.com/prestodb/presto/blob/master/CONTRIBUTING.md#code-style), [development guidelines](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development) and [formatting guidelines](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting)
```
* The link to code style guidelines https://github.com/prestodb/presto#code-style doesn't seem to be correct, so suggest changing it back to https://github.com/prestodb/presto/blob/master/CONTRIBUTING.md#code-style. If I'm missing something here, let me know! 

* Do not use Latin abbreviations. See [GitLab style guide recommended word list entry for &](https://docs.gitlab.com/ee/development/documentation/styleguide/word_list.html#).